soda_machine.rb

```diff
@@ -7,12 +7,19 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.select { |soda| soda.brand == soda_brand } [0]
   end
 
   def sell(soda_brand)
+    selected_soda = sodas.select { |soda| soda.brand == soda_brand } [0]
```

Comment:
As mentioned in the code-through, `#find` is designed for exactly this -- finding the first one (without having to make an array of all the matches and then getting the first one out)